### DISCUSSION TITLE
Can Theme-UI be used in SolidJS?

### DISCUSSION THREAD

**Discussion by thedanchez** [system-ui/theme-ui#2394] [Q&A] [answered]
As the title implies, can ThemeUI be used in a SolidJS app? I would like to build a SolidJS component library and trying to look at what options I can use for theming.

**Comment by hasparus** [comment, marked as answer]
It can be used with React and Preact, and I think we could write some new code, swap imports and force it to work with Solid.
However if you're already using Solid, I think you'd like to have something that gets you the performance of atomic classes.

- I'd choose either utility-classes on their own, like Tailwind, and Solid has the `classList` attribute that makes them way more convenient to use.
- Or [ecsstatic](https://github.com/mayank99/ecsstatic) that can compile away, but gives you Emotion-style DX
- Or [Vanilla Extract](https://vanilla-extract.style/) and [Sprinkles](https://vanilla-extract.style/documentation/packages/sprinkles/) if you don't mind writing your styles in a different file.
